src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -895,4 +963,117 @@ mod tests {
             assert!(output.is_err(), "lhs={:?}, rhs={:?}", lhs, rhs);
         }
     }
+
+    #[test]
+    fn test_real_multiply() {
+        let should_pass = vec![(
+            Real::new(-0.0101001).ok(),
+            Real::new(1.01001).unwrap(),
+            Real::new(-0.01).unwrap(),
+        )];
+
+        for (expected, lhs, rhs) in should_pass {
+            assert_eq!(
+                expected,
+                RpnFnScalarEvaluator::new()
+                    .push_param(lhs)
+                    .push_param(rhs)
+                    .evaluate(ScalarFuncSig::MultiplyReal)
+                    .unwrap()
+            );
+        }
+
+        let should_fail = vec![
+            (
+                Real::new(std::f64::MAX).unwrap(),
+                Real::new(std::f64::MAX).unwrap(),
+            ),
+            (
+                Real::new(std::f64::MAX).unwrap(),
+                Real::new(std::f64::MIN).unwrap(),
+            ),
+        ];
+
+        for (lhs, rhs) in should_fail {
+            assert!(
+                RpnFnScalarEvaluator::new()
+                    .push_param(lhs)
+                    .push_param(rhs)
+                    .evaluate::<Real>(ScalarFuncSig::MultiplyReal)
+                    .is_err(),
+                "{} * {} should fail",
+                lhs,
+                rhs
+            );
+        }
+    }
+
+    #[test]
+    fn test_int_multiply() {
+        let should_pass = vec![(Some(187), 11, 17), (Some(3), -1, -3)];
+        for (expected, lhs, rhs) in should_pass {
+            assert_eq!(
+                expected,
+                RpnFnScalarEvaluator::new()
+                    .push_param_with_field_type(lhs, FieldTypeTp::LongLong)
+                    .push_param_with_field_type(rhs, FieldTypeTp::LongLong)
+                    .evaluate(ScalarFuncSig::MultiplyInt)
+                    .unwrap()
+            );
+        }
+
+        let should_fail = vec![(std::i64::MAX, 2), (std::i64::MIN, -1)];
+        for (lhs, rhs) in should_fail {
+            assert!(
+                RpnFnScalarEvaluator::new()
+                    .push_param_with_field_type(lhs, FieldTypeTp::LongLong)
+                    .push_param_with_field_type(rhs, FieldTypeTp::LongLong)
+                    .evaluate::<Int>(ScalarFuncSig::MultiplyInt)
+                    .is_err(),
+                "{} * {} should fail",
+                lhs,
+                rhs
+            );
+        }
+    }
+
+    #[test]
+    fn test_int_uint_multiply() {
+        let should_pass = vec![(Some(std::i64::MAX), std::i64::MAX, 1), (Some(21), 3, 7)];
```

Comment:
How about `negative signed int * uint` ? This case is missing in previous implementation but is also missing in the new impl.